### ISSUE TITLE
Looks it now working on my gpu？ what wrong？

### ISSUE THREAD

**Issue by EricKong1985** [ggerganov/whisper.cpp#1004]
Dear all,
I am a very new man for c++ coding.
I have clone the project  to my local ubuntu 22 desktop. then install the cuda driver and cuba tookit.
use the command 
make clean
WHISPER_CUBLAS=1 make stream -j
to build the stream binary.
And it pass the compilation and generated the file.
then download the ggml model file from the script.
But after execute stream with sample command :
./stream -m ./models/ggml-base.en.bin -t 8 --step 500 --length 5000
then check the gpu usage by nvilia-smi ,the GPU Utilization is only 1%, then I use top to check cpu ,it is 60%.
and it is decond my voice with very very slow speed,even not response.
What wrong with my operation?
Any guy can help me ?
payment is also negotiable.

eric@eric-Thurley:~/.local/lib/python3.8/site-packages$ nvidia-smi
Mon Jun 12 10:58:49 2023       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                  Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf            Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3070         On | 00000000:04:00.0 Off |                  N/A |
| 53%   47C    P2               42W / 240W|   2012MiB /  8192MiB |      1%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      1118      G   /usr/lib/xorg/Xorg                          156MiB |
|    0   N/A  N/A      1405      G   /usr/bin/gnome-shell                         26MiB |
|    0   N/A  N/A      2551      G   ...713257150,732828957956174509,262144       64MiB |
|    0   N/A  N/A      2886      G   /usr/bin/nvidia-settings                      0MiB |
|    0   N/A  N/A      8563      G   gnome-control-center                          3MiB |
|    0   N/A  N/A     38655      C   ./stream                                    684MiB |
|    0   N/A  N/A     38987      C   ./stream                                    526MiB |
|    0   N/A  N/A     39328      C   ./stream                                    526MiB |
|    0   N/A  N/A     39652      G   ...ures=SpareRendererForSitePerProcess       20MiB |
+---------------------------------------------------------------------------------------+
eric@eric-Thurley:~/.local/lib/python3.8/site-packages$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
Release:	20.04
Codename:	focal

**Comment by bobqianic** [comment]
Fixed in #1472

**Comment by svyatoclav** [comment]
Without good knowledge of all these C++, make, cmake and build stuff I tried 2 days to run `stream` on my NVIDIA GPU. Here is my steps to make this possible to run whisper.cpp stream with CUDA:

```
git clone https://github.com/ggerganov/whisper.cpp.git
cd whisper.cpp
sh ./models/download-ggml-model.sh tiny
cmake -B build -DGGML_CUDA=1
cmake --build build -j --config Release
cd examples
cmake stream -DGGML_CUDA=ON -DCMAKE_CUDA_COMPILER=/opt/cuda/bin/nvcc ..
cd ..
cmake -DWHISPER_SDL2=ON -DGGML_CUDA=ON
make stream
./bin/stream -m ./models/ggml-tiny.bin --step 500 --length 5000
# check with btop/nvtop/nvidia-smi if it runs on GPU with CUDA
```

Hopefully this will help someone in the future!
